### PR TITLE
Adding breadcrumbs

### PR DESCRIPTION
**Pull request by mdbenjam**
Connects: #2364 

Adding a generic way to implement breadcrumbs in our navbar. To auto-generate breadcrumbs you need to do the following:

1) Make either Route or PageRoute objects the children of the NavigationBar.
1) Give each of those Routes/PageRoutes a breadcrumb attribute that has the name of the breadcrumb.
1) Order the routes so that sub pages come after higher level pages.

**Test Plan**
- [ ] Go to the case file and document view page and expect to see the associated breadcrumbs.

### CODE REVIEW

File: client/app/reader/DecisionReviewer.jsx [hunk 1]
```diff
@@ -42,10 +42,6 @@ export class DecisionReviewer extends React.PureComponent {
     history.push(`/${vacolsId}/documents/${docId}`);
   }
 
-  onShowList = (history, vacolsId) => () => {
```
Comment (mdbenjam) [inline client/app/reader/DecisionReviewer.jsx]: Somewhat tangential. I'm changing the one use of this function to a link, which is what it should've been from the beginning.

File: client/app/reader/PdfUI.jsx [hunk 2]
```diff
@@ -165,13 +165,14 @@ export class PdfUI extends React.Component {
     return <div className={pdfUiClass}>
       <div className="cf-pdf-header cf-pdf-toolbar usa-grid-full">
         <span className="usa-width-one-third cf-pdf-buttons-left">
-          { this.props.showClaimsFolderNavigation && <Button
+          { this.props.showClaimsFolderNavigation && <Link
```
Comment (mdbenjam) [inline client/app/reader/PdfUI.jsx]: This is really a link, not a button. Just trying to clean up some unnecessary code and make things more semantic.

File: client/app/components/NavigationBar.jsx [hunk 3]
```diff
@@ -12,7 +30,18 @@ export default class NavigationBar extends React.Component {
       userDisplayName
     } = this.props;
 
-    return <header className="cf-app-header">
+    const breadcrumbComponents = getElementsWithBreadcrumbs(this).map(
+      (route) => <Route path={route.path} render={
+        (props) => <span>
+            <h2 id="page-title" className="cf-application-title">&nbsp; > &nbsp;</h2>
```
Comment (mmlumba) [inline client/app/components/NavigationBar.jsx]: Depending on the app there's different page separators. On Dispatch `|` is the separator, while in Certification/Reader `>` is the separator.

File: client/app/reader/DecisionReviewer.jsx [hunk 4]
```diff
@@ -82,91 +78,77 @@ export class DecisionReviewer extends React.PureComponent {
...
-          <Route
-            path="/:vacolsId/documents"
-            render={this.documentsRoute}
-          />
-          <Route
+          <PageRoute
             exact
             path="/"
-            render={this.routedCaseSelect}
+            title="Assignments | Caseflow Reader"
```
Comment (mmlumba) [inline client/app/reader/DecisionReviewer.jsx]: Does this implementation also take into account pages 3+ levels below the main page? (ie `Reader > some page > sub page`)

Comment (mdbenjam) [reply client/app/reader/DecisionReviewer.jsx]: Yup, you can add an arbitrary number of sub pages. If it matches the URL it'll make it into the breadcrumbs.

File: client/app/components/NavigationBar.jsx (hunk 3 above)
Comment (mdbenjam) [reply client/app/components/NavigationBar.jsx]: To me this is a design issue. There should only be one. @gnakm do you have thoughts on this.

File: client/app/components/NavigationBar.jsx [hunk 5]
```diff
@@ -35,7 +39,9 @@ export default class NavigationBar extends React.Component {
           </div>
         </div>
         <PerformanceDegradationBanner />
-      </header>;
+      </header>
+      {this.props.children}
```
Comment (sharonwarner) [inline client/app/components/NavigationBar.jsx]: Why do we need to add this.props.children here too?

Comment (mdbenjam) [reply client/app/components/NavigationBar.jsx]: To actually display the children. We use them above to calculate the breadcrumbs. But this actually renders them.

**Pull request merged**